lib/generators/shopify_app/scopes_storage/scopes_storage_generator.rb

```diff
@@ -0,0 +1,53 @@
+# frozen_string_literal: true
+require 'rails/generators/base'
+require 'rails/generators/active_record'
+
+module ShopifyApp
+  module Generators
+    class ScopesStorageGenerator < Rails::Generators::Base
+      include Rails::Generators::Migration
+      source_root File.expand_path('../templates', __FILE__)
+
+      class_option :with_cookie_authentication, type: :boolean, default: false
+      class_option :embedded, type: :string, default: 'true'
+
+      def create_scopes_storage_in_shop_model
+        if yes?("Do you want to add scopes column to the Shop model? [y/n]")
+          migration_template('db/migrate/add_scopes_column.erb', 'db/migrate/add_scopes_column.rb')
+          copy_file('shop_with_scopes.rb', 'app/models/shop.rb')
+        end
+      end
+
+      def include_scopes_verification_in_home_controller
+        template('home_controller.rb', 'app/controllers/home_controller.rb') unless with_cookie_authentication?
```

Comment:
Would it be possible to introduce the new concern via a regexp rather than replacing the entire file? It seems to me like this leads us to maintain multiple copies of the home controller, which might be error prone.

Reply:
I just pushed up a change introducing the `scopes` and `scopes=` instance methods for the Shop model.